executor/distsql.go

```diff
@@ -691,9 +720,15 @@ func (e *IndexLookUpExecutor) getResultTask() (*lookupTableTask, error) {
 	if !ok {
 		return nil, nil
 	}
-	err := <-task.doneCh
-	if err != nil {
-		return nil, errors.Trace(err)
+	for err := range task.doneCh {
```

Comment:
Why close the `doneCh` channel and use for loop to receive it?

Reply:
refactored.